### ISSUE TITLE
Response headers not returned on HTTP Requests - Request headers are returned instead [3.4]

### ISSUE THREAD

**Issue by fenix-hub** [godotengine/godot#54669]
### Godot version

3.4.stable

### System information

Windows 11

### Issue description

This bug seems to happen using Godot Engine 3.4 for Windows 11 using the binaries downloaded from https://github.com/godotengine/godot/releases/tag/3.4-stable , for Windows x64.

Http requests made using the `HTTPRequest` node on Godot Engine 3.4 (Windows Platform) are not giving Response headers back from the `request_completed` signal (the `headers` array seems to be empty). 
Instead, the signal is returning **request** headers.

To test this I both tried `request()` and `request_raw()` methods on 3.4.
The same project runned on 3.3.4.stable (without the `request_raw` method) doesn't fail the test.

![snapshot](https://user-images.githubusercontent.com/45271396/140614820-ca430d52-cf62-4fa6-814d-b45401f0c138.png)
*script used*

![image](https://user-images.githubusercontent.com/45271396/140614606-c1402cd2-da8d-412b-a326-841abc4c79ad.png)
*response on 3.3.4 stable*

![image](https://user-images.githubusercontent.com/45271396/140614630-a6b0a4c0-dd27-4819-ad7a-a1ad1a87a6cf.png)
*response on 3.4 stable - notice that no response headers are returned, but only the request header*

### Steps to reproduce

1) Download https://github.com/godotengine/godot/releases/download/3.4-stable/Godot_v3.4-stable_x11.64.zip
2) Make a simple scene with a Control node at the root and an HTTPRequest node as a child
3) Connect the `request_completed` signal and add a simple script to make an http request to a REST endpoint at your choice
4) (*useful*) add a custom header to your request, like `x-my-header = <something>`
5) print `headers` variable's value in the `request_completed` signal callback function
6) run the script

### Minimal reproduction project

[HttpHeadersTest.zip](https://github.com/godotengine/godot/files/7490710/HttpHeadersTest.zip)

This project contains the script showed in the description of this issue

**Comment by Calinou** [comment]
@fenix-hub Can you reproduce this in any of the [3.4 betas and RCs](https://downloads.tuxfamily.org/godotengine/3.4/) to determine when the regression started?

**Comment by fenix-hub** [comment]
@Calinou sure!

(win x64)
- ...
- ✔️ beta5
- ✔️ beta6
- ❌ rc1
- ❌ rc2
- ....

**Comment by fenix-hub** [comment]
Also tested on x11 using Windows Linux Subsystem (Ubuntu 20.04), the issue seems to happen starting from rc1 too.

**Comment by akien-mga** [comment]
Here are the changes between beta 6 and RC 1 (note: needs to be expanded manually to see all commits): https://github.com/godotengine/godot/compare/3e2bb415a9b186596b9ce02debc79590380c2355...90f8cd89a738316563dac9b133628df6bafe2cb2

Most likely culprit seems to be #53800 @Faless.

**Comment by DarkKilauea** [comment]
I think I figured out the issue, there is a typo on line 386 in `http_request.cpp`.

``` cpp
void HTTPRequest::_request_done(int p_status, int p_code, const PoolStringArray &p_headers, const PoolByteArray &p_data) {
	cancel_request();
	emit_signal("request_completed", p_status, p_code, headers, p_data);
}
```

should be:

``` cpp
void HTTPRequest::_request_done(int p_status, int p_code, const PoolStringArray &p_headers, const PoolByteArray &p_data) {
	cancel_request();
	emit_signal("request_completed", p_status, p_code, p_headers, p_data);
}
```

I'll be submitting a PR shortly to fix this once I confirm it addresses the bug and that the bug does not exist in master.

**Comment by Faless** [comment]
Closed via #54683 (and #55012).
